### PR TITLE
Add links to confirmation dialog

### PR DESCRIPTION
**Pull request by AndriiHorishniiMOC**
**Jira issue:**
https://glia.atlassian.net/browse/MOB-2862

**What was solved?**
Added the ability for integrators to add links to their “terms and conditions” or/and “privacy policies” to the live observation confirmation dialog.

**Release notes:**

 - [x] Feature
 - [ ] Ignore
 - [x] Release notes (Is it clear from the description here?)
 - [ ] Migration guide (If changes are needed for integrator already using the SDK - what needs to be communicated? Add underneath please)

**Additional info:**

 - [ ] Tests updated? Added? Unit, acceptance, snapshots?
 - [ ] Logging for future troubleshooting of client issues added?

**Screenshots:**

### CODE REVIEW

**Review by DavDo — CHANGES_REQUESTED** [review]
I don't particularly appreciate adding new parameters to the Option payload and new options view and view-binding buttons that should be used only for a Confirmation dialog, because the options dialog/view is used in several scenarios with different texts. I'd suggest adding a separate view for this dialog with its Payload, ViewBinding, and other components.

File: widgetssdk/src/main/res/values/new_strings.xml [hunk 1]
```diff
@@ -202,9 +202,13 @@
...
-    <string name="live_observation_confirm_title">Before We Continue</string>
-    <string name="live_observation_confirm_message">Please allow the {companyName} representative to view this application screen for improved support.</string>
-    <string name="live_observation_indicator_message">App screen is visible to the agent</string>
+    <string name="engagement_confirm_title">Before We Continue</string>
+    <string name="engagement_confirm_message">Please allow the {companyName} representative to view this application screen for improved support.</string>
+    <string name="engagement_indicator_message">App screen is visible to the agent</string>
+    <string name="engagement_confirm_link1_text">Terms and Conditions</string>
+    <string name="engagement_confirm_link1_url"/>
+    <string name="engagement_confirm_link2_text">Privacy Policies</string>
+    <string name="engagement_confirm_link2_url"/>
```
Comment (DavDo) [inline widgetssdk/src/main/res/values/new_strings.xml:211]: Are these changes submitted to the remote strings?

**Review by andrews-moc — COMMENTED** [review]
I agree with Davit's concerns.

File: widgetssdk/src/main/res/values/new_strings.xml (hunk 1 above)
Comment (AndriiHorishniiMOC) [reply widgetssdk/src/main/res/values/new_strings.xml:211]: PR in review.

**Comment by AndriiHorishniiMOC** [comment]
I added the new dialogue type.

File: widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.kt [hunk 2]
```diff
@@ -517,7 +517,9 @@ class ChatView(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defSty
             theme = theme,
             companyName = companyName,
             positiveButtonClickListener = { onEngagementConfirmationDialogAllowed() },
-            negativeButtonClickListener = { onEngagementConfirmationDialogDismissed() }
+            negativeButtonClickListener = { onEngagementConfirmationDialogDismissed() },
+            link1ClickListener = {}, //TODO: will be implemented on the next task
```
Comment (Pelkar) [inline widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.kt:521]: Can we maybe make this a little more specific? We are expecting these to be ToS and PP right? Maybe we call the listeners accordingly? Would add a lot of useful context to reading the code

File: widgetssdk/src/main/res/values/new_strings.xml [hunk 3]
```diff
@@ -205,6 +205,10 @@
     <string name="live_observation_confirm_title">Before We Continue</string>
     <string name="live_observation_confirm_message">Please allow the {companyName} representative to view this application screen for improved support.</string>
     <string name="live_observation_indicator_message">App screen is visible to the agent</string>
+    <string name="live_observation_confirm_link1_text">Terms and Conditions</string>
+    <string name="live_observation_confirm_link1_url"/>
```
Comment (Pelkar) [inline widgetssdk/src/main/res/values/new_strings.xml:209]: rename to engagement_etc?

**Review by Pelkar — APPROVED** [review]

File: widgetssdk/src/main/res/values/new_strings.xml (hunk 3 above)
Comment (AndriiHorishniiMOC) [reply widgetssdk/src/main/res/values/new_strings.xml]: Updated with renamed strings.

File: widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.kt (hunk 2 above)
Comment (AndriiHorishniiMOC) [reply widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.kt:521]: Actually, these are links one and two. The integrators can use them as they want.
Maybe it makes sense to change it to:
`linkClickListener = { url: String }`

**Review by gugalo — APPROVED** [review]
I have nothing to add

**Review by DavDo — APPROVED** [review]
Great job!
Are you planning to add snapshot tests in a separate task?

**Comment by AndriiHorishniiMOC** [comment]
Yes, adding the snapshot tests is a separate ticket.

**Comment by AndriiHorishniiMOC** [comment]
!squash

**Pull request merged**